PersonaFlow/agentstack
# feature (PersonaFlow-): New file tab and ingest files

## Features
1. The file manager used to live in the Assistant Form. I pulled it out because it is actually a related, but realistically separate form. So you should see this new Files Tab and it should only be selectable if the architecture selected in the Assistant Form supports file uploads.

2. Additionally, we want to support formally ingesting the files - that is, allowing the assistant to actually process the data in the file to be used in future interactions. The idea here is that once a user selects “Upload” in the File Dialog, the /ingest endpoint should be called with that file. What happens next is the /ingest/${taskId}/process should be called which is supposed to display the steamed process of what’s going on “behind the scenes”. This is broken though - so for now, it should just display a loading spinner which on complete, should invalidate the query keys and trigger a refetch of the files (and, displaying the newly ingested file).

To test:
1. Ensure the File tab is only enabled for an architecture that supports files.
2. In the File Tab -> File Dialog, try uploading a new file. You should see a loading spinner followed by a refetch of the files (and then your newly uploaded file in a <Badge /> component)
